tests/courseEditor.test.js

```diff
@@ -5,21 +5,25 @@ const fs = require('fs-extra');
 const path = require('path');
 const async = require('async');
 const ncp = require('ncp');
-const config = require('../lib/config');
 const cheerio = require('cheerio');
+const { exec } = require('child_process');
+const requestp = require('request-promise-native');
+const klaw = require('klaw');
+const tmp = require('tmp');
+
+const config = require('../lib/config');
 const sqldb = require('../prairielib/lib/sql-db');
 const sqlLoader = require('../prairielib/lib/sql-loader');
 const sql = sqlLoader.loadSqlEquiv(__filename);
 const helperServer = require('./helperServer');
-const { exec } = require('child_process');
-const requestp = require('request-promise-native');
-const klaw = require('klaw');
 
 const locals = {};
 let page, elemList;
 
-const baseDir = path.join(__dirname, 'testFileEditor');
-const courseTemplateDir = path.join(baseDir, 'courseTemplate');
+const courseTemplateDir = path.join(__dirname, 'testFileEditor', 'courseTemplate');
+
+// Set up temporary writeable directories for course content
+const baseDir = tmp.dirSync().name;
```

Comment:
Previously, this and `tests/fileEditor` expected to have exclusive use of the directories in `testFileEditor`, which is an invalid assumption when running tests in parallel. We now avoid this problem by giving each test its own temporary directory.

I think there are some `.gitignore` rules that we can now remove.

Reply:
This is the best change ever! A horrible side-effect of the way things used to be was that killing a test half-way through would leave the temporary directories in place, which would then stop the tests from running next time. I'm so happy to see this fixed!